Mathlib/NumberTheory/Padics/PadicVal.lean

```diff
@@ -542,12 +542,34 @@ theorem range_pow_padicValNat_subset_divisors' {n : ℕ} [hp : Fact p.Prime] :
 #align range_pow_padic_val_nat_subset_divisors' range_pow_padicValNat_subset_divisors'
 
 /-- The `p`-adic valuation of `(p * n)!` is `n` more than that of `n!`. -/
-theorem padicValNat_factorial_mul {p : ℕ} (n : ℕ) (hp : p.Prime):
+theorem padicValNat_factorial_mul {p : ℕ} (n : ℕ) [hp : Fact p.Prime]:
     padicValNat p (p * n) ! = padicValNat p n ! + n :=  by
   refine' PartENat.natCast_inj.mp _
-  rw [padicValNat_def' (Nat.Prime.ne_one hp) <| factorial_pos (p * n), Nat.cast_add,
-      padicValNat_def' (Nat.Prime.ne_one hp) <| factorial_pos n]
-  exact Prime.multiplicity_factorial_mul hp
+  rw [padicValNat_def' (Nat.Prime.ne_one hp.out) <| factorial_pos (p * n), Nat.cast_add,
+      padicValNat_def' (Nat.Prime.ne_one hp.out) <| factorial_pos n]
+  exact Prime.multiplicity_factorial_mul hp.out
+
+/-- The `p`-adic valuation of `m` equals zero if it is between `n * k` and `n * (k + 1)` for
+some `k`. -/
+theorem padicValNat_eq_zero_between_consec_multiples {m p k : ℕ}
+    (h1 : p * k < m) (h2 : m < p * (k + 1)) : padicValNat p m = 0 :=
+  padicValNat.eq_zero_of_not_dvd <| not_dvd_of_between_consec_multiples h1 h2
```

Comment:
I think this is a job for `Set.Ioo`:
```suggestion
theorem padicValNat_eq_zero_of_mem_Ioo {m p k : ℕ}
    (hm : m ∈ Set.Ioo (p * k) (p * (k + 1))) : padicValNat p m = 0 :=
  padicValNat.eq_zero_of_not_dvd <| not_dvd_of_between_consec_multiples hm.1 hm.2
```

Reply:
Nice, should we also change `not_dvd_of_between_consec_multiples` in the same way?

Reply:
Yes I think we should; can you do it as part of this PR?

Reply:
done

Reply:
Seems like this introduces some dependency problems...